distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -1420,6 +1420,85 @@ type Table
         Helpers.ensure_same_connection "table" [self, lookup_table] <|
             Lookup_Query_Helper.build_lookup_query self lookup_table key_columns add_new_columns allow_unmatched_rows on_problems
 
+    ## ALIAS find replace
+       GROUP Standard.Base.Calculations
+       ICON join
+       Replaces values in `column` using `lookup_table` to specify a
+       mapping from old to new values.
+
+       Arguments:
+       - lookup_table: the table to use as a mapping from old to new values. A
+         `Map` can also be used here (in which case `from_column` and
+         `to_column` are ignored).
+       - column: the column within `self` to perform the replace on.
+       - from_column: the column within `lookup_table` to match against `column`
+         in `self`.
+       - to_column: the column within `lookup_table` to get new values from.
```

Comment:
I'd really like to avoid
> (in which case `from_column` and `to_column` are ignored

IMO having arguments that only make sense in one context but are to be ignored in another is not great design. I remember we used to try really hard to avoid issues like that in the initial designs for `Table`, like `select_columns` back in the day.

What if we replace this with `lookup_table` being `Lookup | Map`, where
```
type Lookup
    Table table (from : Integer | Text) (to : Integer | Text)
```

Then, in 'table' mode we'd write:
```
src_table.replace (Lookup.Table lookup "from" "to")
```
and if we do a map:
```
src_table.replace map
```
so there's no ignored arguments like now.

---

A downside is you have to create the nested `Lookup` atom. This can be done in the GUI by widgets, but it is still a bit problematic as then just dragging a lookup table onto it will not work (until we have 'smart' from conversions that somehow allow to add more arguments whenever an implicit conversion from `Table` to `Lookup` happens).

---

This downside is quite problematic - so I guess maybe we should just keep it as is... Just wanted to think through, because if possible, I think it would be good to avoid such ignored arguments.

Reply:
Yes, this is definitely worth thinking through. The approach in enso is to have methods that do things in multiple ways, often with parameters that are only meaningful in some cases, so this will be keep being a question.

The 'smart' conversion could perhaps work like this:

```
type Lookup
    Table table (from : Integer | Text) (to : Integer | Text)
    Map (map: Map)

Lookup.from (table:Table) = Table.table # arguments are not supplied
```

This would be a non-standard way to use `from` since the type would be (Table -> Integer | Text -> Integer | Text -> Lookup).

Reply:
Should it be an error to supply a `Map` and also `from_column` or `to_column`? I think maybe it should, so that would change "ignored" to "not allowed".

Reply:
> The approach in enso is to have methods that do things in multiple ways, often with parameters that are only meaningful in some cases, so this will be keep being a question.

Is it? I think we try to avoid it whenever possible. What methods already do this?



> Should it be an error to supply a `Map` and also `from_column` or `to_column`? I think maybe it should, so that would change "ignored" to "not allowed".

Yes, I think that is a good idea.

Reply:
> > The approach in enso is to have methods that do things in multiple ways, often with parameters that are only meaningful in some cases, so this will be keep being a question.
> 
> Is it? I think we try to avoid it whenever possible. What methods already do this?

I guess what I should have said is: I have seen this before (I don't remember where) and I expect we will be tempted again. It's definitely something we should avoid if we can.

Reply:
Now throws an error if from/to_column is used with a Map.